core/meterer/dynamodb_metering_store.go

```diff
@@ -241,18 +241,18 @@ func (s *DynamoDBMeteringStore) GetPeriodRecords(ctx context.Context, accountID
 			":reservationPeriod": &types.AttributeValueMemberN{Value: strconv.FormatUint(reservationPeriod, 10)},
 		},
 		ScanIndexForward: aws.Bool(true),
-		Limit:            aws.Int32(MinNumBins),
+		Limit:            aws.Int32(limit),
 	}
 	bins, err := s.dynamoClient.QueryWithInput(ctx, queryInput)
 	if err != nil {
-		return [MinNumBins]*pb.PeriodRecord{}, fmt.Errorf("failed to query payments for account: %w", err)
+		return make([]*pb.PeriodRecord, MinNumBins), fmt.Errorf("failed to query payments for account: %w", err)
```

Comment:
should this be length of `limit` instead of `MinNumBins`? also for below ones

Reply:
A valid query w/limit 1000 might only return 100 valid results - in which case it returns 100 PeriodRecords. But if it doesn't find anything, it would then return 1000 empty PeriodRecords - that seems odd. My original impl returned empty slice, but when I refactored GetPeriodRecords, it always returned `MinNumBins` size slice - which I don't totally understand why. So I chose to not break any caller assumptions and return `MinNumBins` size slice when the query fails.

Reply:
I see! It was for the clients to always be able to create the MinNumBins even if they are empty. the reasoning for dataapi makes sense to me